cocos/2d/CCNode.cpp

```diff
@@ -57,7 +57,12 @@ NS_CC_BEGIN
 
 bool nodeComparisonLess(Node* n1, Node* n2)
 {
-    return(n1->_localZOrder.value < n2->_localZOrder.value);
+#if defined(_M_X64) || defined(_LP64) || defined(__x86_64) || defined(_WIN64)
```

Comment:
The android is based on linux, so I think linux 64 specific macros can handle it.